sql/core/src/main/scala/org/apache/spark/sql/artifact/ArtifactManager.scala

```diff
@@ -258,9 +315,10 @@ class ArtifactManager(session: SparkSession) extends Logging {
    * Returns a [[ClassLoader]] for session-specific jar/class file resources.
    */
   def classloader: ClassLoader = {
-    val urls = getAddedJars :+ classDir.toUri.toURL
+    val urls = (getAddedJars :+ classDir.toUri.toURL).toArray
     val prefixes = SparkEnv.get.conf.get(CONNECT_SCALA_UDF_STUB_PREFIXES)
     val userClasspathFirst = SparkEnv.get.conf.get(EXECUTOR_USER_CLASS_PATH_FIRST)
+    val fallbackClassLoader = session.sharedState.jarClassLoader
```

Comment:
This classloader is used only in Hive to hold all `ADD JAR`s.

This classloader will be set to default once there's one ADD JAR: https://github.com/apache/spark/blob/6a36c43604a638f603a3a40c22ee1e6bd3ae8d7e/sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala#L188

Reply:
mmm'kay... not loving it....